crates/polars-io/src/lib.rs

```diff
@@ -167,7 +167,7 @@ pub(crate) fn finish_reader<R: ArrowReader>(
 }
 
 static CLOUD_URL: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"^(s3|gs|gs|gcs|file|abfs|abfss|azure|az|adl)\:\/\/.+$").unwrap());
```

Comment:
Why is the `\:\/\/.+$` sequence no longer required?

Reply:
So `://` didn't need to be escaped. What about `.+$`? It used to be a regex that would match the whole URL, now it just matches a prefix.

Reply:
As for the `.+$`: a rust-match just checks if the regex is present in the passed string. But you just want to check that it starts with any of the given cloud provider options.
The only thing that is different now is that you could -theoretically- pass in `s3://` and it will match, whereas before it wouldn't match.
However, if you passed in `s3://a`, it would fail later on anyhow because it's not a valid url. So, I figured that in either case it'd start to fail as soon as you go to the object_store crate anyhow. And that this regex was only intended to check "starts with".

Reply:
I guess `is_match` still works. All right then! This should be a bit more optimal.